cypress/integration/pages/testsForAllPages.js

```diff
@@ -63,14 +63,151 @@ export const testsThatFollowSmokeTestConfigforAllPages = ({
+              'have.attr',
+              'content',
+              appConfig[service].defaultImageAltText,
+            );
+            cy.get('meta[name="twitter:image:src"]').should(
+              'have.attr',
+              'content',
+              appConfig[service].defaultImage,
+            );
+            cy.get('meta[name="twitter:site"]').should(
+              'have.attr',
+              'content',
+              appConfig[service].twitterSite,
+            );
           });
         });
       }
 
+      const runDescriptionAndTitleTests = ({ description, title }) => {
+        const pageTitle = `${title} - ${appConfig[service].brandName}`;
+
+        cy.get('head').within(() => {
+          cy.title().should('eq', pageTitle);
+          cy.get('meta[name="og:description"]').should(
+            'have.attr',
+            'content',
+            description,
+          );
+          cy.get('meta[name="og:title"]').should(
+            'have.attr',
+            'content',
+            pageTitle,
+          );
+          cy.get('meta[name="twitter:description"]').should(
+            'have.attr',
+            'content',
+            description,
+          );
+          cy.get('meta[name="twitter:title"]').should(
+            'have.attr',
+            'content',
+            pageTitle,
+          );
+        });
+      };
+
+      it('should have correct title & description metadata', () => {
+        /*
+         * Naidheachdan needs to have correct metadata added to all environments.
+         * Naidheachdan condition will be removed in issue https://github.com/bbc/simorgh/issues/
+         */
+        if (pageType !== 'errorPage404' && service !== 'naidheachdan') {
+          cy.request(`${config[service].pageTypes[pageType].path}.json`).then(
+            ({ body }) => {
+              let description;
+              let title;
+              switch (pageType) {
+                case 'articles':
+                  description =
+                    body.promo.summary || body.promo.headlines.seoHeadline;
+                  title = body.promo.headlines.seoHeadline;
+                  break;
+                case 'frontPage':
+                  description = body.metadata.summary;
+                  title = body.promo.name;
+                  break;
+                case 'liveRadio':
+                  description = body.promo.summary;
+                  title = body.promo.name;
+                  break;
+                default:
+                  description = '';
+                  title = '';
+              }
+              runDescriptionAndTitleTests({ description, title });
+            },
+          );
+        }
+
+        if (pageType === 'errorPage404') {
+          runDescriptionAndTitleTests({
+            description: appConfig[service].translations.error[404].title,
+            title: appConfig[service].translations.error[404].title,
+          });
+        }
```

Comment:
I decided to have the metadata tests here, instead of in the `errorPage404` directory, since it is easier to maintain when all pagetype metadata tests are in one place, so if they need to be updated, they can be updated for all pagetypes.

Reply:
This precedent seems like it could lead to a lack of maintainability across all tests that is worse than the improvement in maintainability by this here.

Reply:
Okay, so if I move this into `cypress/integration/pages/errorPage404/tests.js`  and keep the other pagetype tests here in this file, where would you recomment I move these ~26 lines of repeated code? https://github.com/bbc/simorgh/pull/3364/files/58c2645a0d1684f4fd4038960d2af46674a951b8#diff-6ef73c82b1834f188b8648a2ce247f32R142

If we just duplicate - the error pages will get easily forgotten and go stale.

Reply:
After discussion offline, have moved error page tests into `errorPage404/tests.js`